src/lib/constants.ts

```diff
@@ -14,19 +14,29 @@
  * limitations under the License.
  */
 
-export const PRE_UA = 0x1;
-export const ON_UA = 0x2;
-export const ONCE = 0x4;
+export module ClientAb {
 
-export const OPERATIONS = {
-  OP_CUSTOM_JS: 0,
-  OP_INSERT_BEFORE: 1,
-  OP_INSERT_AFTER: 2,
-  OP_PREPEND: 3,
-  OP_APPEND: 4,
-  OP_REPLACE: 5,
-  OP_SET_INNERHTML: 6,
-  OP_REMOVE: 7,
-  OP_SET_ATTRIBUTE: 8,
-  OP_REDIRECT: 9
-};
\ No newline at end of file
+  export type Transform = [
+    flag: Flags, selector: string, operation: Operations, ...rest: any
+  ]
+
+  export const enum Flags {
+    Pre_UA = 1 << 0,
```

Comment:
My understanding is that TypeScript convention is for enum values to be PascalCase? 
```suggestion
    PreUA = 1 << 0,
```

And down below,
```
  CustomJs = 0,
  InsertBefore = 1,
  InsertAfter = 2,
  ...
```

Reply:
Updated!